src/sqlfluff/core/linter.py

```diff
@@ -907,27 +980,41 @@ def parse_string(
         bencher("Finish parsing {0!r}".format(short_fname))
         return ParsedString(parsed, violations, time_dict, templated_file, config)
 
-    @staticmethod
-    def extract_ignore_from_comment(comment: RawSegment):
```

Comment:
Split this function into two parts:
* `extract_ignore_from_comment()` takes a `RawSegment` and in turn calls `parse_noqa()`
* `parse_noqa()` takes a string (the comment text).

Splitting the code this way simplifies unit testing.